### DISCUSSION TITLE
Renovate CI job fails due to timeout

### DISCUSSION THREAD

**Discussion by mbrandau** [renovatebot/renovate#8488] [Request Help]
Hi there 👋🏼

I'm running on-prem Renovate 24.32.3 on my on-prem GitLab. Since 24.27.4 I have quite a lot of the jobs failing due to a 1h limit. With `top` I saw a process `git-remote-http` with 100% CPU. After killing that process the job just continued. Since the runner which executes renovate and GitLab are on the same server I don't know if this process belongs to GitLab or renovate.

Did anyone else encounter this issue? I have never seen that behaviour prior to 24.27.4.
Any help is appreciated!

Edit:
now renovate exited with the following error:
```
DEBUG: Renovate exiting
 INFO: Renovate is exiting with a non-zero code due to the following logged errors
       "loggerErrors": [
         {
           "name": "renovate",
           "level": 50,
           "logContext": "**********",
           "repository": "*********",
           "err": {
             "task": {
               "commands": [
                 "fetch",
                 "origin",
                 "refs/heads/renovate/*:refs/remotes/origin/renovate/*",
                 "--depth=2",
                 "--force"
               ],
               "format": "utf-8"
             },
             "message": "error: git-remote-https died of signal 15\nfatal: expected flush after ref listing\n",
             "stack": "Error: error: git-remote-https died of signal 15\nfatal: expected flush after ref listing\n\n    at GitExecutorChain.onFatalException (/usr/src/app/node_modules/simple-git/src/lib/runners/git-executor-chain.ts:67:77)\n    at GitExecutorChain.<anonymous> (/usr/src/app/node_modules/simple-git/src/lib/runners/git-executor-chain.ts:59:21)\n    at Generator.throw (<anonymous>)\n    at rejected (/usr/src/app/node_modules/simple-git/src/lib/runners/git-executor-chain.js:6:65)\n    at processTicksAndRejections (internal/process/task_queues.js:93:5)"
           },
           "msg": "Repository has unknown error"
         }
       ]
```

Some of the last debug messages after which it seems to stall:
`DEBUG: Committing files to branch renovate/xxx (repository=xxx/xxx, branch=renovate/xxx)`
`DEBUG: Initializing git repository into /builds/ops/renovate/renovate/repos/gitlab/xxx/xxx (repository=xxx/xxx)`
`DEBUG: Setting current branch to master (repository=xxx/xxx)`

**Comment by HonkingGoose** [comment]
The `24.x.x` releases of Renovate are no longer supported, please try the latest version of Renovate (currently `25.28.2`).
Let us know if the newer version fixes the problem or not.

**Reply by mbrandau** [reply]
Hi there,

we‘re currently running `25.63.1` in Kubernetes following these instructions: https://docs.renovatebot.com/self-hosting/#kubernetes, https://docs.renovatebot.com/self-hosting/#kubernetes-for-gitlab-using-git-over-ssh.
Our GitLabs SSH is running on a different port than 22. The SSH config mounted to the renovate container was changed to match the port.
However I‘m still encountering the issue described above.

**Reply by HonkingGoose** [reply]
Thank you for retrying on a newer version of Renovate. 😄 

I don't know enough to help, so you'll have to wait for a maintainer to help debug the problem.

**Comment by viceice** [comment]
Can you try our official helm chart and use http instead of ssh for git?

https://github.com/renovatebot/helm-charts

**Reply by viceice** [reply]
@rakins @HonkingGoose  i think  we should add our helm chart as preferred version for Kubernetes and remove the other Kubernetes samples.

**Reply by HonkingGoose** [reply]
@viceice I've copy/pasted your comment into PR #11202, as it makes no sense to edit the files on the `main` branch as that will just conflict with the big rewrite of the getting started docs. 😄

**Reply by mbrandau** [reply]
I didn‘t even know about the helm chart. I‘ll check it out, but it was still using http instead of ssh. SSH is currently not supported for GitLab. (https://github.com/renovatebot/renovate/issues/5406#issuecomment-797621880)
I managed to get SSH to work on a fork though. But still this is more or less a workaround for this problem. However GitLab could also be the culprit here.

**Reply by viceice** [reply]
Feel free to comment on the ssh issue and suggest a solution if you know something helpful 🙃

**Reply by mbrandau** [reply]
I created a draft to add an option to hostRules to use SSH with GitLab: #11221
